### PR TITLE
Add authorization service and add basic-authorization lambda to impor…

### PR DESCRIPTION
**Pull request by tsialiukigor**
### **Task 7 (Lambda Authorizer + Cognito Authorization)**

All requirements are done. The estimated score is 6.

Client application url: https://d2w2h6xfwejks9.cloudfront.net/
localStorage key: authorization_token, value: c2lqbHZlcjpURVNUX1BBU1NXT1JE 

FE implementation: https://github.com/sijlver/nodejs-aws-fe/commit/0b485fb7f0991a649fc6b72d9cfc73dba35f995b

### CODE REVIEW

File: authorization-service/src/handlers/basic-authorizer.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  try {
+    const { authorizationToken, methodArn } = event;
+    const encodedCredentials = authorizationToken.split(' ')[1];
+    const buffer = Buffer.from(encodedCredentials, 'base64');
+    const [ username, password ] = buffer.toString('utf-8').split(':');
+
+    console.log(`Username: ${username}, Password: ${password}`);
+
+    const effect = username === TEST_USER || password === TEST_PASSWORD ? 'Allow' : 'Deny';
```
Comment (AlexanderFraltsov) [inline authorization-service/src/handlers/basic-authorizer.ts]: That is, if only the username is correct, authorization will still pass (I checked, it is so). Use && instead ||

File: import-service/serverless.ts [hunk 2]
```diff
@@ -64,6 +94,13 @@ const serverlessConfiguration: Serverless = {
                   name: true
                 }
               }
+            },
+            authorizer: {
+              name: 'basicAuthorizer',
+              arn: '${cf:authorization-service-dev.basicAuthorizerArn}',
```
Comment (AlexanderFraltsov) [inline import-service/serverless.ts:100]: '${cf:authorization-service-${self:provider.stage}.basicAuthorizerArn}' - why not?

File: authorization-service/src/handlers/basic-authorizer.ts (hunk 1 above)
Comment (KarneyenkaDzmitry) [reply authorization-service/src/handlers/basic-authorizer.ts:26]: image
It is working even the authorization token is wrong. Please fix and reply.

Comment (tsialiukigor) [reply authorization-service/src/handlers/basic-authorizer.ts]: Updated

Comment (tsialiukigor) [reply authorization-service/src/handlers/basic-authorizer.ts]: It worked so as I wanted to check either login or password (it was for testing perspective). When you add smt at the end of the token the validation for login worked and it allowed you to proceed with this validation. Anyway, I updated and now it works as you expected

Comment (KarneyenkaDzmitry) [reply authorization-service/src/handlers/basic-authorizer.ts]: Great. Now I can set you the 6 grade without any objections

**Pull request merged**